ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/regression/LogisticRegression.java

```diff
@@ -0,0 +1,160 @@
+import org.tribuo.classification.LabelFactory;
+import org.tribuo.classification.sgd.LabelObjective;
+import org.tribuo.classification.sgd.linear.LinearSGDTrainer;
+import org.tribuo.classification.sgd.objectives.Hinge;
+import org.tribuo.classification.sgd.objectives.LogMulticlass;
+import org.tribuo.math.StochasticGradientOptimiser;
+import org.tribuo.math.optimisers.AdaGrad;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Function(FunctionName.LOGISTIC_REGRESSION)
+public class LogisticRegression implements Trainable, Predictable {
+
+    private static final LogisticRegressionParams.ObjectiveType DEFAULT_OBJECTIVE_TYPE = LogisticRegressionParams.ObjectiveType.LOGMULTICLASS;
+    private static final LogisticRegressionParams.OptimizerType DEFAULT_OPTIMIZER_TYPE = LogisticRegressionParams.OptimizerType.ADA_GRAD;
+    private static final double DEFAULT_LEARNING_RATE = 1.0;
+
+    //AdaGrad, AdaDelta, AdaGradRDA, Adam, RMSProp
+    private static final double DEFAULT_EPSILON = 0.1;
+    private static final int DEFAULT_EPOCHS = 5;
+    private static final int DEFAULT_LOGGING_INTERVAL = 1000;
+    private static final int DEFAULT_BATCH_SIZE = 1;
+    private static final Long DEFAULT_SEED = Trainer.DEFAULT_SEED;
+
+    private int epochs;
+    private int loggingInterval;
+    private int minibatchSize;
+    private long seed;
+
+    private LogisticRegressionParams parameters;
+    private StochasticGradientOptimiser optimiser;
+    private LabelObjective objective;
+    /**
+     * Initialize a linear regression algorithm.
+     * @param parameters the parameters for linear regression algorithm
+     */
+    public LogisticRegression(MLAlgoParams parameters) {
+        this.parameters = parameters == null ? LogisticRegressionParams.builder().build() : (LogisticRegressionParams)parameters;
+        validateParameters();
+        createObjective();
+        createOptimiser();
+    }
+
+    private void validateParameters() {
+        if (parameters.getLearningRate() != null && parameters.getLearningRate() < 0) {
+            throw new IllegalArgumentException("Learning rate should not be negative.");
+        }
+
+        if (parameters.getEpsilon() != null && parameters.getEpsilon() < 0) {
+            throw new IllegalArgumentException("Epsilon should not be negative.");
+        }
+
+        if (parameters.getEpochs() != null && parameters.getEpochs() < 0) {
+            throw new IllegalArgumentException("Epochs should not be negative.");
+        }
+
+        if (parameters.getBatchSize() != null && parameters.getBatchSize() < 0) {
+            throw new IllegalArgumentException("MiniBatchSize should not be negative.");
+        }
+
+        if (parameters.getLoggingInterval() != null && parameters.getLoggingInterval() < 0) {
+            throw new IllegalArgumentException("Logging intervals should not be negative.");
```

Comment:
From java doc of `org.tribuo.classification.sgd.linear.LinearSGDTrainer`
> Log the loss after this many iterations. If -1 don't log anything.

Reply:
Good catch!